libs/plugins/shared.mb_str_replace.php

```diff
@@ -44,7 +44,8 @@ function smarty_mb_str_replace($search, $replace, $subject, &$count = 0)
                 }
             }
         } else {
-            $parts = mb_split(preg_quote($search), $subject);
+            $parts = mb_split(preg_quote($search, null), $subject);
+            $parts = $parts ?: array();
```

Comment:
you could add `?: array()` to the end of the previous line as well

Reply:
yes, it will be nicer